third_party/2and3/atomicwrites/__init__.pyi

```diff
@@ -1,13 +1,13 @@
-from typing import AnyStr, Callable, ContextManager, IO, Optional, Text, Type
+from typing import Any, AnyStr, Callable, ContextManager, Generic, IO, Optional, Text, Type, Union
 
 def replace_atomic(src: AnyStr, dst: AnyStr) -> None: ...
 def move_atomic(src: AnyStr, dst: AnyStr) -> None: ...
-class AtomicWriter(object):
-    def __init__(self, path: AnyStr, mode: Text = ..., overwrite: bool = ...) -> None: ...
-    def open(self) -> ContextManager[IO]: ...
-    def _open(self, get_fileobject: Callable) -> ContextManager[IO]: ...
-    def get_fileobject(self, dir: Optional[AnyStr] = ..., **kwargs) -> IO: ...
-    def sync(self, f: IO) -> None: ...
-    def commit(self, f: IO) -> None: ...
-    def rollback(self, f: IO) -> None: ...
-def atomic_write(path: AnyStr, writer_cls: Type[AtomicWriter] = ..., **cls_kwargs: object) -> ContextManager[IO]: ...
+class AtomicWriter(Generic[AnyStr]):
+    def __init__(self, path: Union[Text, bytes], mode: Text = ..., overwrite: bool = ...) -> None: ...
+    def open(self) -> ContextManager[IO[AnyStr]]: ...
+    def _open(self, get_fileobject: Callable[..., IO[AnyStr]]) -> ContextManager[IO[AnyStr]]: ...
+    def get_fileobject(self, dir: Optional[AnyStr] = ..., **kwargs: Any) -> IO[AnyStr]: ...
+    def sync(self, f: IO[AnyStr]) -> None: ...
+    def commit(self, f: IO[AnyStr]) -> None: ...
+    def rollback(self, f: IO[AnyStr]) -> None: ...
+def atomic_write(path: Union[Text, bytes], writer_cls: Type[AtomicWriter[AnyStr]] = ..., **cls_kwargs: object) -> ContextManager[IO[AnyStr]]: ...
```

Comment:
Here I'd guess the return value is mostly determined by mode being passed through `cls_kwargs`, not by the `writer_cls` argument. This could presumably be done correctly with a lot of overloads, but I'm not sure it's worth the trouble.